cobalt/android/apk/app/src/main/java/dev/cobalt/util/AssetLoader.java

```diff
@@ -0,0 +1,29 @@
+package dev.cobalt.util;
+
+import static dev.cobalt.util.Log.TAG;
+
+import android.content.Context;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.io.IOException;
+
+/** Utility functions for read asset. */
+public class AssetLoader {
+
+    private AssetLoader() {}
+
+    public static String loadJavaScriptFromAssets(Context context, String filename) {
+        try {
+            InputStream is = context.getAssets().open(filename);
+            int size = is.available();
+            byte[] buffer = new byte[size];
+            is.read(buffer);
```

Comment:
Maybe check the actual bytes read?

Reply:
revamped this for handling large js files.